File: engine/engine-model/src/main/scala/tools/packet/quest/info/QuestFinish.scala
package tools.packet.quest.info

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class QuestFinish(private var _questId: Short, private var _npcId: Int, private var _nextQuestId: Short) extends PacketInput {
  def questId: Short = _questId

  def npcId: Int = _npcId

  def nextQuestId: Short = _nextQuestId

  override def opcode(): SendOpcode = SendOpcode.UPDATE_QUEST_INFO
}

File: engine/engine-model/src/main/java/tools/packet/quest/info/QuestFinish.java
package tools.packet.quest.info;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record QuestFinish(Short questId, Integer npcId, Short nextQuestId) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.UPDATE_QUEST_INFO;
   }
}